util/src/main/scala/org/ensime/util/path.scala

```diff
@@ -0,0 +1,29 @@
+package org.ensime.util
+
+import java.net.{ JarURLConnection, URL }
+import java.nio.charset.Charset
+import java.nio.file.{ Path => JPath }
+
+import com.google.common.io.Files
+/**
+ * Created by harsh on 15/06/16.
+ */
+package object path {
+  type Path = JPath
+  implicit class EnrichPath(val path: Path) extends AnyVal {
+    def isJarUrl = path.toString.startsWith("jar:file:")
+    def isJava: Boolean = path.toString.endsWith(".java")
+    def isScala: Boolean = path.toString.endsWith(".scala")
+    def readString(implicit cs: Charset): String = {
+      if (!path.isJarUrl) Files.toString(path.toFile, cs) else {
+        val in = new URL(path.toString).openConnection().asInstanceOf[JarURLConnection].getInputStream
```

Comment:
I did a bit of findings and hit and trial and this is how it worked. Also isn't `in.close()` closing the jar stream? I will get rid of `scala.io.Source` though .